### DISCUSSION TITLE
OIDC and having two issuers where one acts as the middle man

### DISCUSSION THREAD

**Discussion by rukshanfonseka** [walt-id/.github#57] [Help] [answered]
I wanted to check can Walt.Id out of the box support this scenario?

A wallet presents a VP to us using OIDC. We can then validate the credential and perform other necessary checks and then if all is good redirect to another Issuer using OIDC who will then prompt for further info to generate the actual VCs.

**Comment by waltkb** [comment, marked as answer]
Like this?

1. Wallet presents credential(s) to a Verifier
2. Verifier runs various verification policies over presented credentials
3. (policies passed) Verifier sends issuance request to Issuer
4. Issuer responds with issuance request start link
5. Verifier sends issuance request start link to user
6. (user clicks issuance request start link) User can redeem credential(s) from the Issuer to their wallet

We have implemented this flow multiple times for clients, but it is not something that comes out-of-the-box (as it is very specific). However, you can easy implement this using the walt.id Wallet Kit.
